src/cascadia/TerminalControl/TSFInputControl.cpp

```diff
@@ -0,0 +1,379 @@
+        _textBlock = Controls::TextBlock();
+        _textBlock.Visibility(Visibility::Collapsed);
+        _textBlock.IsTextSelectionEnabled(false);
+        _textBlock.TextDecorations(TextDecorations::Underline);
+
+        // Canvas for controlling exact position of the TextBlock
+        _canvas = Windows::UI::Xaml::Controls::Canvas();
+        _canvas.Visibility(Visibility::Collapsed);
+
+        // add the Textblock to the Canvas
+        _canvas.Children().Append(_textBlock);
+
+        // set the content of this control to be the Canvas
+        this->Content(_canvas);
+
+        // Create a CoreTextEditingContext for since we are acting like a custom edit control
+        auto manager = Core::CoreTextServicesManager::GetForCurrentView();
+        _editContext = manager.CreateEditContext();
+
+        // sets the Input Pane display policy to Manual for now so that it can manually show the
+        // software keyboard when the control gains focus and dismiss it when the control loses focus.
+        // TODO investigate if we can set this to Automatic
+        _editContext.InputPaneDisplayPolicy(Core::CoreTextInputPaneDisplayPolicy::Manual);
+
+        // set the input scope to Text because this control is for any text.
+        _editContext.InputScope(Core::CoreTextInputScope::Text);
+
+        _editContext.TextRequested({ this, &TSFInputControl::_textRequestedHandler });
+
+        _editContext.SelectionRequested({ this, &TSFInputControl::_selectionRequestedHandler });
+
+        _editContext.FocusRemoved({ this, &TSFInputControl::_focusRemovedHandler });
+
+        _editContext.TextUpdating({ this, &TSFInputControl::_textUpdatingHandler });
+
+        _editContext.SelectionUpdating({ this, &TSFInputControl::_selectionUpdatingHandler });
+
+        _editContext.FormatUpdating({ this, &TSFInputControl::_formatUpdatingHandler });
+
+        _editContext.LayoutRequested({ this, &TSFInputControl::_layoutRequestedHandler });
+
+        _editContext.CompositionStarted({ this, &TSFInputControl::_compositionStartedHandler });
+
+        _editContext.CompositionCompleted({ this, &TSFInputControl::_compositionCompletedHandler });
+    }
+
+    // Method Description:
+    // - NotifyFocusEnter handler for notifying CoreEditTextContext of focus enter
+    //   when TerminalControl receives focus.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TSFInputControl::NotifyFocusEnter()
+    {
+        if (_editContext != nullptr)
+        {
+            _editContext.NotifyFocusEnter();
+        }
+    }
+
+    // Method Description:
+    // - NotifyFocusEnter handler for notifying CoreEditTextContext of focus leaving.
+    //   when TerminalControl no longer has focus.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TSFInputControl::NotifyFocusLeave()
+    {
+        if (_editContext != nullptr)
+        {
+            //_editContext.NotifyFocusLeave();

```

Comment:
Leave an explanation or delete the code

Reply:
I've created an issue #3645 to track if it's needed